kirin/core/merge_utils.py

```diff
@@ -264,26 +264,146 @@ def yield_next_stop_from_trip_update(new_trip_update, db_trip_update):
 
 
 def yield_next_stop_from_base_schedule_vj(navitia_vj):
-    # Iterate on the theoretical VJ if the new trip update doesn't list all stop_times
+    # Iterate on the base-schedule VJ if the new trip update doesn't list all stop_times
     for order, vj_st in enumerate(navitia_vj.get("stop_times", [])):
         yield order, vj_st
 
 
-def is_past_midnight(prev_stop_event, next_stop_event):
-    if prev_stop_event.time is None or next_stop_event.time is None:
+def convert_nav_stop_list_to_stu_list(nav_stop_list, circulation_date):
+    """
+    Convert navitia's json stop list to dated StopTimeUpdate list
+    :param nav_stop_list: list of dict containing navitia' stop info (extracted from json)
+    :param circulation_date: date of the first stop time event of the list
+    :return: A list of not impacted StopTimeUpdates
+    """
+    stus = []
+    previous_stop_dep_time = datetime.time.min
+    for nav_order, nav_stop in enumerate(nav_stop_list):
+        current_stop_arr_time = nav_stop.get("utc_arrival_time")
+        if is_past_midnight(previous_stop_dep_time, current_stop_arr_time):
+            circulation_date += datetime.timedelta(days=1)
+        current_stop_arr_dt = datetime.datetime.combine(circulation_date, current_stop_arr_time)
+
+        current_stop_dep_time = nav_stop.get("utc_departure_time")
+        if is_past_midnight(current_stop_arr_time, current_stop_dep_time):
+            circulation_date += datetime.timedelta(days=1)
+        current_stop_dep_dt = datetime.datetime.combine(circulation_date, current_stop_dep_time)
+
+        stu = StopTimeUpdate(
+            nav_stop.get("stop_point"),
+            arrival=current_stop_arr_dt,
+            departure=current_stop_dep_dt,
+            order=nav_order,
+        )
+        stus.append(stu)
+    return stus
+
+
+def is_past_midnight(previous_hour, current_hour):
+    if previous_hour is None or current_hour is None:
         return False
+    return previous_hour > current_hour
+
 
+def is_past_midnight_event(prev_stop_event, next_stop_event):
```

Comment:
next_stop_event is ambiguous
```suggestion
def is_past_midnight_event(prev_stop_event, current_stop_event):
```

Reply:
Done in https://github.com/CanalTP/kirin/pull/354/commits/86e4194716aa46d1db9c9c5d15ad5f85d588afb9